executor/show.go

```diff
@@ -1346,6 +1346,11 @@ func (e *ShowExec) fetchShowCreateUser() error {
 			fmt.Sprintf("'%s'@'%s'", e.User.Username, e.User.Hostname))
 	}
 
+	authplugin := ""
+	if len(rows) == 1 {
+		authplugin = rows[0].GetString(0)
+	}
+
```

Comment:
I think this should read something like:
```suggestion
	authplugin := mysql.AuthNativePassword
	if len(rows) == 1 && rows[0].GetString(0) != "" {
		authplugin = rows[0].GetString(0)
	}

```

There is a potential upgrade issue where even though the bootstrap task will add the column for plugin, and convert all existing users to have the value `mysql_native_password`, because of mixed TiDB-server versions, older version servers may create empty-value plugin entries. In which defaulting to `mysql_native_password` might be the best way to handle it.